### PR TITLE
Reorganize legacy charmrepo code.

### PR DESCRIPTION
**Pull request by frankban**
This branch only includes moved code.
Move current charmstore impl to legacy.go, and the local repository impl to local.go.
New charmstore code will be implemented in a future charmstore.go file.

### CODE REVIEW

File: charmrepo/legacy.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
```
Comment (rogpeppe) [inline charmrepo/legacy.go]: // Package charmrepo implements access to charm repositories.

?

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by rogpeppe** [comment]
LGTM with one trivial suggestion

File: charmrepo/legacy.go (hunk 1 above)
Comment (frankban) [inline charmrepo/legacy.go]: Done.

**Pull request merged**